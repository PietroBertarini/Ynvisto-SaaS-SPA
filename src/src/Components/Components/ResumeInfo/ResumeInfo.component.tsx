import React from 'react';
import Typography from '@material-ui/core/Typography';
import Box from '@material-ui/core/Box';

import { useStyles } from './ResumeInfo.styles';

export enum EResumeStatus {
    GOOD = 'GOOD',
    WARNING = 'WARNING',
    BAD = 'BAD',
}

export interface ResumoInfoProps {
    /**
     * Value of
     */
    value: number
    /**
     * Subtitle Text of ResumeInfo
     */
    subtitle: string;
    /**
     * Status of resume will be represent by a color,
     * NoStatus - White,
     * Alert - Yellow,
     * Warning - Red,
     * Ok - Green.
     */
    status: EResumeStatus;

}
/**
 * Simple component that have a value in R$ format with status color and a subtitle.
 */
function ResumeInfo(props : ResumoInfoProps) {
  const {
    value, subtitle,
  } = props;
  const classes = useStyles(props);
  return (
    <Box display="flex" flexDirection="column" alignSelf="center" justifyContent="center">
      <Typography className={classes.resumeHeader}>
        {value.toLocaleString('pt-br', { style: 'currency', currency: 'BRL' })}
      </Typography>
      <Typography className={classes.resumeSubTitle} align="center">
        {subtitle}
      </Typography>
    </Box>

  );
}
export default ResumeInfo;
