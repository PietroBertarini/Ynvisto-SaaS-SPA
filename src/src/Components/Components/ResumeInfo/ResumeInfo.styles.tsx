import { makeStyles, createStyles } from '@material-ui/core/styles';

const styledBy = (property, mapping) => (props) => mapping[props[property]];

export const useStyles = makeStyles(() => createStyles({
  resumeSubTitle: {
    color: '#B1B1B1',
    fontSize: '1.5vw',
    textAlign: 'center',

  },
  resumeHeader: {
    textAlign: 'center',
    color: styledBy('status', {
      GOOD: '#05E985',
      WARNING: '#E9C46A',
      BAD: '#EB5050',
    }),
    fontSize: '2.8vw',
  },
}));
