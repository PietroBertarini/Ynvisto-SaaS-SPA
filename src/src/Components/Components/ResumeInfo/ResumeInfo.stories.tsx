import React from 'react';
// also exported from '@storybook/react' if you can deal with breaking changes in 6.1
import { Meta, Story } from '@storybook/react/types-6-0';
import ResumeInfo, { EResumeStatus, ResumoInfoProps } from './ResumeInfo.component';

export default {
  title: 'Components/Resume/SubComponents/ResumeInfo',
  component: ResumeInfo,
} as Meta;

const Template: Story<ResumoInfoProps> = (args) => <ResumeInfo {...args} />;
export const Warning = Template.bind({});
Warning.args = {
  value: 30_043.02,
  subtitle: 'Legenda',
  status: EResumeStatus.BAD,
};

export const Alert = Template.bind({});
Alert.args = {
  value: 30_043.02,
  subtitle: 'Legenda',
  status: EResumeStatus.WARNING,
};
export const Ok = Template.bind({});
Ok.args = {
  value: 30_043.02,
  subtitle: 'Legenda',
  status: EResumeStatus.GOOD,
};
