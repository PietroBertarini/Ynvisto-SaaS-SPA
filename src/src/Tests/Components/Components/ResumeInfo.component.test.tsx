import React from 'react';
import { shallow } from 'enzyme';
import ResumeInfo, { EResumeStatus, ResumoInfoProps } from '../../../Components/Components/ResumeInfo/ResumeInfo.component';

test('Renders ResumoInfo components and match to previously snapshot', () => {
  const mockProps : ResumoInfoProps = {
    value: 30_304.94,
    subtitle: 'Maio',
    status: EResumeStatus.GOOD,
  };

  const defaultStatusComponent = shallow(
    <ResumeInfo
      status={EResumeStatus.GOOD}
      value={mockProps.value}
      subtitle={mockProps.subtitle}
    />,
  );
  const alertStatusComponent = shallow(
    <ResumeInfo
      status={EResumeStatus.WARNING}
      value={mockProps.value}
      subtitle={mockProps.subtitle}
    />,
  );
  const warningStatusComponent = shallow(
    <ResumeInfo
      status={EResumeStatus.BAD}
      value={mockProps.value}
      subtitle={mockProps.subtitle}
    />,
  );
  const okStatusComponent = shallow(
    <ResumeInfo
      status={EResumeStatus.GOOD}
      value={mockProps.value}
      subtitle={mockProps.subtitle}
    />,
  );
  expect(defaultStatusComponent).toMatchSnapshot();
  expect(alertStatusComponent).toMatchSnapshot();
  expect(warningStatusComponent).toMatchSnapshot();
  expect(okStatusComponent).toMatchSnapshot();
});
