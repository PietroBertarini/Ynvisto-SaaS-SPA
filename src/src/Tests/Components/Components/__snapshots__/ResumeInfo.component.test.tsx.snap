// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders ResumoInfo components and match to previously snapshot 1`] = `
<Styled(MuiBox)
  alignSelf="center"
  display="flex"
  flexDirection="column"
  justifyContent="center"
>
  <WithStyles(ForwardRef(Typography))
    className="makeStyles-resumeHeader-2 makeStyles-resumeHeader-3"
  >
    R$ 30.304,94
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Typography))
    align="center"
    className="makeStyles-resumeSubTitle-1"
  >
    Maio
  </WithStyles(ForwardRef(Typography))>
</Styled(MuiBox)>
`;

exports[`Renders ResumoInfo components and match to previously snapshot 2`] = `
<Styled(MuiBox)
  alignSelf="center"
  display="flex"
  flexDirection="column"
  justifyContent="center"
>
  <WithStyles(ForwardRef(Typography))
    className="makeStyles-resumeHeader-2 makeStyles-resumeHeader-4"
  >
    R$ 30.304,94
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Typography))
    align="center"
    className="makeStyles-resumeSubTitle-1"
  >
    Maio
  </WithStyles(ForwardRef(Typography))>
</Styled(MuiBox)>
`;

exports[`Renders ResumoInfo components and match to previously snapshot 3`] = `
<Styled(MuiBox)
  alignSelf="center"
  display="flex"
  flexDirection="column"
  justifyContent="center"
>
  <WithStyles(ForwardRef(Typography))
    className="makeStyles-resumeHeader-2 makeStyles-resumeHeader-5"
  >
    R$ 30.304,94
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Typography))
    align="center"
    className="makeStyles-resumeSubTitle-1"
  >
    Maio
  </WithStyles(ForwardRef(Typography))>
</Styled(MuiBox)>
`;

exports[`Renders ResumoInfo components and match to previously snapshot 4`] = `
<Styled(MuiBox)
  alignSelf="center"
  display="flex"
  flexDirection="column"
  justifyContent="center"
>
  <WithStyles(ForwardRef(Typography))
    className="makeStyles-resumeHeader-2 makeStyles-resumeHeader-6"
  >
    R$ 30.304,94
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Typography))
    align="center"
    className="makeStyles-resumeSubTitle-1"
  >
    Maio
  </WithStyles(ForwardRef(Typography))>
</Styled(MuiBox)>
`;
